### ISSUE TITLE
System.Exception: 'The socket is connected, you needn't connect again!'

### ISSUE THREAD

**Issue by ooples** [alpacahq/alpaca-trade-api-csharp#248]
I'm getting this exception with running the new extensions reconnect method and so far from what I can tell it seems to happen almost everytime the reconnect method is running. I have sent you an email with more info and the code to run

I'm receiving an error saying unknown server protocol version after the socket disconnects then the reconnect method tries to do its thing and then I receive this error

```cs
System.Exception
  HResult=0x80131500
  Message=The socket is connected, you needn't connect again!
  Source=SuperSocket.ClientEngine
  StackTrace:
   at SuperSocket.ClientEngine.TcpClientSession.Connect(EndPoint remoteEndPoint)
   at WebSocket4Net.WebSocket.Open()
   at WebSocket4Net.WebSocket.<OpenAsync>d__2.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Alpaca.Markets.StreamingClientBase`1.<ConnectAndAuthenticateAsync>d__17.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at Alpaca.Markets.Extensions.ClientWithReconnectBase`2.<handleSocketClosed>d__23.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by OlegRa** [comment]
I've found an issue in the existing reconnection logic. In fact, it will try to reconnect even if the first reconnection attempt will be successful. I'll commit fix and publish an updated version of packages later after more testing on my side.

**Comment by OlegRa** [comment]
@ooples  You can test the fix in version [3.9.0](https://github.com/alpacahq/alpaca-trade-api-csharp/releases/tag/v3.9.0) available on NuGet.

**Comment by ooples** [comment]
I'm not seeing issues today but I will reopen if I see any bugs related to this
